lib/storage/file.js

```diff
@@ -1173,6 +1173,8 @@ File.prototype.getSignedUrl = function(options, callback) {
   }[options.action];
 
   var name = encodeURIComponent(this.name);
+  var targetGeneration = is.undefined(this.generation) ?
+    false : encodeURIComponent(this.generation);
```

Comment:
Sorry for the nitpick, but could we move some of this logic around line 1226? I think we should wait to encode the generation until after `getCredentials` succeeds.